spec/ean/ean_spec.rb

```diff
@@ -0,0 +1,26 @@
+require "spec_helper"
+
+describe BrDocuments::Ean do
+  describe "#valid?" do
+    it "is invalid with length different to 8, 13 and 14" do
+      ["123456189", "114567854871"].each do |code|
+        ean = described_class.new(code)
+        expect(ean).to_not be_valid
+      end
+    end
+
+    it "is invalid with invalid check number" do
+      ["12345611", "1145678548712", "1234567891127"].each do |number|
+        ean = described_class.new(number)
+        expect(ean).to_not be_valid
+      end
+    end
+
```

Comment:
context "When the checksum is valid"
      context "When the EAN has 8 digits"
        it { is_expected.to be_valid }
      context "When the EAN has 13 digits"
        it { is_expected.to be_valid }
      context "When the EAN has 14 digits"
        it { is_expected.to be_valid }
      context "When the number of digits is different of 8, 13 or 14"
        it { is_expected.to_not be_valid }